### DISCUSSION TITLE
Error when running IBM i Code Coverage test

### DISCUSSION THREAD

**Discussion by devinhh** [codefori/vscode-ibmi#804] [IBM i]
When I try to run a Code Coverage test I get the following error:

Commands:
		CODECOV CMD(CALL DTDEVIN/OCEAN2022) MODULE((DTDEVIN/OCEAN2022 *PGM *ALL)) OUTSTMF('/tmp/1660314084718.cczip')

CPF1992:  STRDBG command not allowed in thread capable job.
CPF0001:  Error found on STRCCA command.
CPF0006:  Errors occurred in command.
CPD0039:  Command ENDCCA not valid in this mode.
CPF0001:  Error found on ENDCCA command.
CPF0006:  Errors occurred in command.
CPF2407:  Message file QRSEMSGS in *LIBL not found.
CEE9901:  Application error.  CPF2407 unmonitored by CODECOV at statement 0000000003, instruction X'0000'.

**Comment by worksofliam** [comment]
* What OS version are you on?
* Please add `QDEVTOOLS` to your library list and try again!

**Reply by devinhh** [reply]
7.4
I added QDEVTOOLS and got the same message:

Current library: QGPL
Library list: QTEMP GITDEVIN FAL2206PF TSTLIB2206 DMSSRV2206 DMSBNDDIR EISGLOBAL EISFUSION EIS TST2206PF EISQTEMP EISSRC ENSIGN EISTOOL ALLXML100 QGPL DBUCUR SVN RPGUNIT HTTPAPI RJSERS PROFOUNDUI XAOBJ13216 XAPROD DTDEVIN_1 DTDEVINUTL ENTMOVE EISTRGLIB GITWRK QSHONI DTDEVIN QDEVTOOLS
Commands:
		CODECOV CMD(CALL DTDEVIN/OCEAN2022) MODULE((DTDEVIN/OCEAN2022 *PGM *ALL)) OUTSTMF('/tmp/1660314609265.cczip')

CPF1992:  STRDBG command not allowed in thread capable job.
CPF0001:  Error found on STRCCA command.
CPF0006:  Errors occurred in command.
CPD0039:  Command ENDCCA not valid in this mode.
CPF0001:  Error found on ENDCCA command.
CPF0006:  Errors occurred in command.
CCR0024:  Code coverage failed.

**Reply by worksofliam** [reply]
@devinhh Can you share the output from the `Code for IBM i` output channel? Thanks!

**Reply by devinhh** [reply]
Current library: QGPL
Library list: QTEMP GITDEVIN FAL2206PF TSTLIB2206 DMSSRV2206 DMSBNDDIR EISGLOBAL EISFUSION EIS TST2206PF EISQTEMP EISSRC ENSIGN EISTOOL ALLXML100 QGPL DBUCUR SVN RPGUNIT HTTPAPI RJSERS PROFOUNDUI XAOBJ13216 XAPROD DTDEVIN_1 DTDEVINUTL ENTMOVE EISTRGLIB GITWRK QSHONI DTDEVIN QDEVTOOLS
Commands:
		CODECOV CMD(CALL DTDEVIN/OCEAN2022) MODULE((DTDEVIN/OCEAN2022 *PGM *ALL)) OUTSTMF('/tmp/1660314609265.cczip')

CPF1992:  STRDBG command not allowed in thread capable job.
CPF0001:  Error found on STRCCA command.
CPF0006:  Errors occurred in command.
CPD0039:  Command ENDCCA not valid in this mode.
CPF0001:  Error found on ENDCCA command.
CPF0006:  Errors occurred in command.
CCR0024:  Code coverage failed.

**Reply by worksofliam** [reply]
@devinhh That is the 'IBM i output', but I need the 'Code for IBM i' output. Thanks!

**Reply by devinhh** [reply]
/home: /QOpenSys/usr/bin/qsh
liblist -d QTEMP EISFORMS EIS_LZFORM DTLF DTDATA EISSRC EISGLOBAL EISFUSION EIS EIS_FINC EIS_CNVGBL EIS_MNFDTA EIS_GBLDTA ENSIGN EISTOOL ALLXML100 QGPL DBUCUR RJSERS RJSSMTP DBUAUD;liblist -c QGPL;liblist -a QDEVTOOLS DTDEVIN QSHONI GITWRK EISTRGLIB ENTMOVE DTDEVINUTL DTDEVIN_1 XAPROD XAOBJ13216 PROFOUNDUI RJSERS HTTPAPI RPGUNIT SVN DBUCUR QGPL ALLXML100 EISTOOL ENSIGN EISSRC EISQTEMP TST2206PF EIS EISFUSION EISGLOBAL DMSBNDDIR DMSSRV2206 TSTLIB2206 FAL2206PF GITDEVIN QTEMP;system  "CODECOV CMD(CALL DTDEVIN/OCEAN2022) MODULE((DTDEVIN/OCEAN2022 *PGM *ALL)) OUTSTMF('/tmp/1660323240614.cczip')"; if [[ $? -ne 0 ]]; then exit 1; fi
{
    "code": 1,
    "signal": null,
    "stdout": "",
    "stderr": "CPF1992:  STRDBG command not allowed in thread capable job.\nCPF0001:  Error found on STRCCA command.\nCPF0006:  Errors occurred in command.\nCPD0039:  Command ENDCCA not valid in this mode.\nCPF0001:  Error found on ENDCCA command.\nCPF0006:  Errors occurred in command.\nCCR0024:  Code coverage failed."
}

**Reply by worksofliam** [reply]
@devinhh Something to do with your setup.. let me get back to you. Same command works on my system

```
/home/linux/builds/company_system: /QOpenSys/usr/bin/qsh
liblist -d QGPL QTEMP QDEVELOP QBLDSYS QBLDSYSR SOTEST;liblist -c LIAMA;liblist -a QDEVTOOLS SAMPLE CMPSYS;system  "CODECOV CMD(CALL CMPSYS/TEST) MODULE((CMPSYS/TEST *PGM *ALL)) OUTSTMF('/tmp/1660330702473.cczip')"; if [[ $? -ne 0 ]]; then exit 1; fi
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": "CPC7301:  File CODECOV created in library QTEMP.\nCPC7305:  Member CCA0001 added to file CODECOV in QTEMP.\nCPI433A:  Unable to retrieve query options file.\nCPI433A:  Unable to retrieve query options file.\nCPI433A:  Unable to retrieve query options file.\nCCR0018:  Module (TEST) is instrumented.\nCCR0005:  Code coverage session has ended.\nCPI433A:  Unable to retrieve query options file.\nCPI433A:  Unable to retrieve query options file.\nQSH0005:  Command ended normally with exit status 0.\nCPC7309:  Member CCA0001 removed from file CODECOV in QTEMP.\nCCR0021:  Created code coverage result file (/tmp/1660330702473.cczip)."
}
```

**Comment by devinhh** [comment]
![Screenshot 2022-08-12 122601](https://user-images.githubusercontent.com/35005044/184421013-e5cee3e1-7de1-44fb-9704-54f994f6dd3f.png)
I am also seeing this message at times.

**Comment by worksofliam** [comment]
@devinhh Please do a `DSPJOBD` on the JOBD for your user profile and look for `Allow multiple threads` and share the value here. Thanks!

**Reply by devinhh** [reply]
![image](https://user-images.githubusercontent.com/35005044/184429230-07eae606-157a-4307-9266-e917cf9392ed.png)

**Reply by devinhh** [reply]
@worksofliam Were you able to see what I might be missing?

**Reply by worksofliam** [reply]
@devinhh no, sorry. It's something related to your job configuration and multi threaded jobs.

**Reply by worksofliam** [reply]
@devinhh I have carved out some time later in the week to look into this.

**Reply by devinhh** [reply]
@worksofliam Do you have any ideas on this?

**Reply by worksofliam** [reply]
Hey @devinhh - no, sorry. I just got back from some holiday time and I am playing catch up for a while.

**Reply by devinhh** [reply]
I am still trying to get this to work.

**Comment by Balrocj** [comment]
Hi, I'm having the same error. Is there any way to fix it? I checked my user's jobd and the parameter is set to No.

**Reply by worksofliam** [reply]
Is this error happening with the IBM i Testing extension?

cc @SanjulaGanepola: this is an old thread, but new question

**Reply by SanjulaGanepola** [reply]
I believe I saw this message as well when `QDEVTOOLS` was not on my library list. If you are using the IBM i Testing extension, please share the full error from the output channel as described [here](https://codefori.github.io/docs/developing/testing/troubleshooting/#extension-logs).
